src/com/mambu/apisdk/services/LoansService.java

```diff
@@ -1181,6 +1184,36 @@ public LoanTransaction executeJSONTransactionRequest(String accountId, LoanTrans
 
 	}
 
+	/**
+	 * Convenience method for updating/changing the interest rate on a loan account.
+	 * 
+	 * @param accountId
+	 *            the id of the account.
+	 * @param transactionType
+	 *            loan transaction type. Must not be null. Supported type is: INTEREST_RATE_CHANGED
+	 * @param transactionRequest
+	 *            JSON transaction request
+	 * 
+	 * @return loan transaction
+	 * 
+	 * @throws MambuApiException
+	 */
+	public LoanTransaction postInterestRateChange(String accountId, LoanTransactionType transactionType,
+			JSONTransactionRequest transactionRequest) throws MambuApiException {
+
+		if (transactionRequest == null || transactionType == null) {
+			throw new IllegalArgumentException("Transaction request and transactionType must not be null");
+		}
+
+		if (!transactionType.equals(INTEREST_RATE_CHANGED)) {

```

Comment:
Since the only purpose of this method is to change the interest rate of a loan account, it would be better to remove the "transactionType" argument from the method signature and to specify it behind the scenes, when calling the "executeJSONTransactionRequest" method. That way we can also get rid of this "if" statement.